liblocation/src/main/java/com/mapbox/android/core/location/AndroidLocationEngineImpl.java

```diff
@@ -141,16 +89,19 @@ public void removeLocationUpdates(@NonNull LocationListener listener) {
   }
 
   @Override
-  public void removeLocationUpdates(@NonNull PendingIntent pendingIntent) {
+  public void removeLocationUpdates(PendingIntent pendingIntent) {
```

Comment:
Well technically if you annotate this with `NonNull` then you wouldn't need to null check, i'm removing this annotation just to silence lint and do a null check.

Reply:
On this note, why do we accept `null` values here at all if it's a no-op in such case?

Reply:
@LukasPaczos what do you mean by "accept"? wouldn't annotation just result in warning in IDE? The main reason i removed annotation is to silence the ide warnings. also i do null check here because i wanted wrapper to behave consistently, since passing `null` value to `LocationManager` vs. `FusedLocationProviderClient` will result in different error.

Reply:
That's totally fine! I just wanted to point this out as I'm not a fun of null-checking whole code blocks that are in our control (not invoked from outside libraries, that we are unsure of the implementation) as this can hide programmatic errors.